### PR TITLE
[12.0][FIX] chained_swapper: Avoid singleton error to update records

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa

### CODE REVIEW

File: chained_swapper/models/chained_swapper.py [hunk 1]
```diff
@@ -89,7 +89,9 @@ def _onchange_model_id(self):
     def write(self, vals):
         super().write(vals)
```
Comment (pedrobaeza) [inline chained_swapper/models/chained_swapper.py]:
```suggestion
        res = super().write(vals)
```
and return at the end

File: chained_swapper/models/chained_swapper.py [hunk 2]
```diff
@@ -89,7 +89,9 @@ def _onchange_model_id(self):
...
         if 'name' in vals:
-            self.ref_ir_act_window_id.name = self.name
+            for rec in self:
+                if self.ref_ir_act_window_id:
+                    self.ref_ir_act_window_id.name = self.name
```
Comment (pedrobaeza) [inline chained_swapper/models/chained_swapper.py]:
```suggestion
            self.filtered("ref_ir_act_window_id").write({"name": vals["name"]})
```

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by sergio-teruel** [comment]
@pedrobaeza Changes done!!

File: chained_swapper/models/chained_swapper.py (hunk 1 above)
Comment (pedrobaeza) [reply chained_swapper/models/chained_swapper.py]: Please do this also (not critical in this case, but good habit for avoiding side effects)

File: chained_swapper/models/chained_swapper.py [hunk 3]
```diff
@@ -89,7 +89,7 @@ def _onchange_model_id(self):
     def write(self, vals):
         super().write(vals)
         if 'name' in vals:
-            self.ref_ir_act_window_id.name = self.name
+            self.filtered("ref_ir_act_window_id").write({"name": vals["name"]})
```
Comment (carlosdauden) [inline chained_swapper/models/chained_swapper.py]:
```suggestion
            self.mapped("ref_ir_act_window_id").write({"name": vals["name"]})
```

Comment (pedrobaeza) [inline chained_swapper/models/chained_swapper.py]: Sorry for my mistake. It's:

```suggestion
            self.mapped("ref_ir_act_window_id").write({"name": vals["name"]})
```

**Review by carlosdauden — CHANGES_REQUESTED** [review]

**Comment by sergio-teruel** [comment]
done

**Review by pedrobaeza — APPROVED** [review]
Thanks for the changes

**Review by carlosdauden — APPROVED** [review]

**Review by HaraldPanten — APPROVED** [review]

**Pull request merged**